engine/plugins/server/server.js

```diff
@@ -120,6 +138,38 @@ module.exports = function(options, imports, register) {
             });
         });
 
+        /**
+         * Token has been refreshed in the frontend.
+         *
+         * Change token for client.
+         */
+        socket.on('TokenRefreshed', (data) => {
+            const previousToken = token;
+
+            token = data.token;
+            fetch(options.tokenValidationEndPoint + token).then(res => res.json()).then(data => {
+                // Validate the token and send error if not valid.
+                if (Object.prototype.hasOwnProperty.call(data, 'valid') && !data.valid) {
+                    socket.emit('error', { message: 'Not authorized', code: 401 });
+                    socket.disconnect(true);
+                    return;
+                }
+
+                // Set that current token is valid.
+                isTokenValid = true;
```

Comment:
Should the token have been `isTokenValid = false;` in the `socket.on('RefreshToken',` ?

Reply:
But it is actually not invalid... because the refresh is an hour before expire.